### PR TITLE
BaSP-TG: Added create task

### PR DESCRIPTION
**Pull request by LucianaPrat**
The method to add a task is added. If the fields are empty or one is missing, the task cannot be created.
[localhost:3000/tasks/add](http://localhost:3000/tasks/add)

### CODE REVIEW

File: src/resources/tasks.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    taskName,
+    description: taskDescription,
+  };
+
+  if (!taskName && !taskDescription) {
+    res.status(400).send({ success: false, msg: 'You must add name and description' });
+  } else if (!taskName) {
+    res.status(400).send({ success: false, msg: 'You must add a name' });
+  } else if (!taskDescription) {
+    res.status(400).send({ success: false, msg: 'You must add a description' });
```
Comment (silveriac) [inline src/resources/tasks.js:26]: Description may be optional

Comment (nicolasblasco) [reply src/resources/tasks.js:26]: I agree

File: src/resources/tasks.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import express from 'express';
+
+const fs = require('fs');
+
+const router = express.Router();
+
+const tasks = require('../data/tasks.json');
+
+router.post('/add', (req, res) => {
```
Comment (nicolasblasco) [inline src/resources/tasks.js:9]: The route should be / because the method already specifies the action of creating

File: src/resources/tasks.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import express from 'express';
+
+const fs = require('fs');
```
Comment (nicolasblasco) [inline src/resources/tasks.js:3]: We should use import for libraries

File: src/resources/tasks.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+const router = express.Router();
+
+const tasks = require('../data/tasks.json');
+
+router.post('/add', (req, res) => {
+  const task = req.body;
+  const { taskName } = task;
+  const taskDescription = task.description;
```
Comment (nicolasblasco) [inline src/resources/tasks.js:12]: We could directly destructure the task like this: { taskName, taskDescription } = task

File: src/resources/tasks.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  const ids = tasks.map((t) => t.id);
+  const maxId = Math.max(...ids);
+  const newTask = {
+    id: maxId + 1,
+    taskName,
+    description: taskDescription,
+  };
+
+  if (!taskName && !taskDescription) {
+    res.status(400).send({ success: false, msg: 'You must add name and description' });
```
Comment (nicolasblasco) [inline src/resources/tasks.js:22]: We should use res.JSON()

File: src/resources/tasks.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  } else if (!taskName) {
+    res.status(400).send({ success: false, msg: 'You must add a name' });
```
Comment (nicolasblasco) [inline src/resources/tasks.js:24]: Same as above

File: src/resources/tasks.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  } else if (!taskDescription) {
+    res.status(400).send({ success: false, msg: 'You must add a description' });
+  } else {
+    tasks.push(newTask);
+  }
+
+  fs.writeFile('src/data/tasks.json', JSON.stringify(tasks), (err) => {
+    if (err) {
+      res.status(500).send({ success: false, msg: 'Cannot save new task' });
```
Comment (nicolasblasco) [inline src/resources/tasks.js:33]: Same as above and the status code should be 400

File: src/resources/tasks.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    } else {
+      res.status(201).send({ success: true, msg: 'Task Created', data: tasks });
```
Comment (nicolasblasco) [inline src/resources/tasks.js:35]: Same as above

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**